### PR TITLE
convienience constructors for 3D rotations

### PR DESCRIPTION
**Pull request by jabooth**
Expands the convenience constructors we already have for 2D Rotations to 3D, specifically for rotations around the X, Y, and Z axes.

TODO: add tests.

### CODE REVIEW

File: menpo/transform/homogeneous/rotation.py [hunk 1]
```diff
@@ -151,6 +151,84 @@ def init_3d_from_quaternion(cls, q):
...
+            theta = theta * np.pi / 180.0
+        return Rotation(np.array([[ np.cos(theta), 0, np.sin(theta)],
+                                  [             0, 1,             0],
+                                  [-np.sin(theta), 0, np.cos(theta)]]),
+                        skip_checks=True)
+
+    @classmethod
+    def init_from_3d_ccw_angle_around_z(cls, theta, degrees=True):
+        r"""
+        Convenience constructor for 3D CCW rotations around the z axis
```
Comment (patricksnape) [inline menpo/transform/homogeneous/rotation.py]: Can we maybe just state for these what the handedness convention is, which way is up etc? So people can know how to orient their meshes?

File: menpo/transform/homogeneous/rotation.py [hunk 2]
```diff
@@ -151,6 +151,84 @@ def init_3d_from_quaternion(cls, q):
...
+            interpreted as radians.
+
+        Returns
+        -------
+        rotation : :map:`Rotation`
+            A 3D rotation transform.
+        """
+        if degrees:
+            # convert to radians
+            theta = theta * np.pi / 180.0
```
Comment (patricksnape) [inline menpo/transform/homogeneous/rotation.py]: https://docs.scipy.org/doc/numpy/reference/generated/numpy.deg2rad.html

File: menpo/transform/homogeneous/rotation.py (hunk 1 above)
Comment (jabooth) [inline menpo/transform/homogeneous/rotation.py]: Not sure what you mean? `ccw` infers Counter ClockWise, so to me it seems fully specified. I guess we could state that XYZ is a RH coordinate system but it feels that this assumption is made in far more places than just this in menpo...

P.S. think it's a little dangerous to state what 'up' is - I guess it doesn't really have a meaning here. If I have a facial mesh where `+z` is up, then to spin the head around I would use to `init_from_3d_ccw_angle_around_z` method. If the mesh is orientated with `+y` as up, I'd use the `init_from_3d_ccw_angle_around_y` to spin that one in a CCW fashion...unless I'm missing your meaning here?

File: menpo/transform/homogeneous/rotation.py (hunk 2 above)
Comment (jabooth) [inline menpo/transform/homogeneous/rotation.py]: Fair cop!

**Comment by patricksnape** [comment]
Jenkins rebuild this please

**Comment by patricksnape** [comment]
Jenkins retest this please

**Comment by jabooth** [comment]
@patricksnape all passed - any objections or can we bring it in?

**Comment by patricksnape** [comment]
Just tests my main man

**Pull request merged**